### PR TITLE
set up GitHub Actions to have a simple cI system

### PR DESCRIPTION
**Pull request by szabgab**

### CODE REVIEW

**Comment by hirooih** [comment]
@szabgab 
Thank you. 
I just migrated this repository from SourceForge. to GitHub last weekend.  This is my first original repository on GitHub. I am not familiar with Docker, either. I have been at a loss by seeing a lot of documents on https://docs.github.com/en/actions.

Your PR is a great clue for me. Give me some time to understand your `ci.yml`. 
The most important thing I learned was knowning that I could run `apt-get` to install GNU Readline Library.
And it seems that I can use `brew `command on macOS.

> https://github.com/k6io/example-github-actions/blob/master/.github/workflows/macos.yml

Let me ask you some questions.

***

What is the difference between two containers you used, `perldocker/perl-tester` and `perl`?
Could you give the pointer to `perl` container as https://hub.docker.com/r/perldocker/perl-tester for `perldocker/perl-tester`?

***

I don't think we need the following lines;

```
  schedule:
    - cron: '42 5 * * *'
```

Am I correct?

***

Do we need to add `workflow_dispatch` to test a workflow manually?

Thanks in advance.

**Comment by szabgab** [comment]
Recently I was trying various ways to install Term::ReadLine::Gnu because it was a dependency of one of the modules I was testing. You can see the results here: https://github.com/szabgab/testing-Term-ReadLine-Gnu/
Feel free to pick up anything from there.

I set up an extensive CI system for Test::Class that you can see here: https://github.com/szabgab/test-class
There is some explanation in the yaml file (including why I have the scheduled job), but if you have time you might even want to watch the video where I explain the whole thing:  https://perlmaven.com/github-actions-ci-job-for-test-class

I don't know if  https://hub.docker.com/r/perldocker/perl-tester as a full link would work, but  https://hub.docker.com/ is the default location to look for images. The full URL might also work. Why would you want it?

You need workflow_dispatch if you'd like to trigger the job manually and not just using push.

**Comment by szabgab** [comment]
For our purposes on GitHub Actions a Docker image is just a VM with some stuff already installed, but you can use the regular apt/yum commands that are appropriate for the packaging system of the distribution that was used to create the image.

**Comment by hirooih** [comment]
Thank you for your reply.  I will check links you provided.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '42 5 * * *'
```
Comment (hirooih) [inline .github/workflows/ci.yml:7]: Remove these lines.

I read your comments on https://github.com/szabgab/test-class/blob/master/.github/workflows/ci.yml and watch your video https://perlmaven.com/github-actions-ci-job-for-test-class.

cpantesters covers the issue for CPAN modules.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        run: |
+            perl -v
+            #apt-get update
+            apt-get install -y libreadline-dev
+
+      - name: Regular tests
+        run: |
+            perl Makefile.PL
+            make
+            make test
```
Comment (hirooih) [inline .github/workflows/ci.yml:40]: Remove this job.

Term::ReadLine::Gnu uses only standard modules for portability.  We don't need perldocker/perl-tester.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        perl-version:
+          - '5.32'
+          - '5.30'
+          - '5.20'
+
+    container:
+      image: perl:${{ matrix.perl-version }}
```
Comment (hirooih) [inline .github/workflows/ci.yml:56]: Add a comment of this image, for example '# https://hub.docker.com/_/perl'.

Do they support '5.20'?  It seems that only '5.28', '5.30', and '5.32' are supported.

**Review by hirooih — CHANGES_REQUESTED** [review]
Thank you for your PR. This is what I was looking for.  I am studying GitHub Actions. Tips you gave me help me a lot.

I would like to ask you to make this workflow simple as far as possible as I commented.

I want to use GitHub Actions to check a wrong commit.
[cpantesters](http://matrix.cpantesters.org/?dist=Term-ReadLine-Gnu+1.40) test my module periodically on various Perl versions. We don't have do same things here.

Thanks.

**Comment by szabgab** [comment]
I don't think you are right, but I am sure you can now create your own configuration.

**Comment by hirooih** [comment]
> I don't think you are right, but I am sure you can now create your own configuration.

I see.
I will create my configuration based on your PR.

Again, thank you for your valuable advices.

**Pull request closed without merging**